w1/assignment4_1/e4/main.cpp

```diff
@@ -0,0 +1,21 @@
+#define NDEBUG
+#include <cassert>
+#include <iostream>
+#include <string>
+
+int divide(int dividend, int divisor)
+{
+    assert(divisor != 0 && "Division by zero not allowed");
+    return dividend/divisor;
+}    
+
+
+int main() 
+{
+    assert(divide(4,2) && "division by zero not allowed");
+    std::cout << "Test 1 passed" << "\n";
+
+    assert(divide(4,0) && "division by zero not allowed");
```

Comment:
This will never be caught here, as the assert inside the function will catch the assertion before this is executed 😊